src/Swarm/Game/Step.hs

```diff
@@ -671,6 +672,53 @@ execConst c vs k = do
         robotInventory .= inv'
         return $ Out VUnit k
       _ -> badConst
+    Drill -> case vs of
+      [VDir d] -> do
+        rname <- use robotName
+        inv <- use robotInventory
+        loc <- use robotLocation
+        rDir <- use robotOrientation
+
+        let nextLoc = loc ^+^ applyTurn d (rDir ? V2 0 0)
+        em <- lift . lift $ use entityMap
+        drill <- lookupEntityName "drill" em `isJustOr` cmdExn Drill ["Drill does not exist?!"]
+        nextE <- entityAt nextLoc >>= (`isJustOr` cmdExn Drill ["There is nothing to drill", "in the direction", "of robot", rname <> "."])
+        -- TODO: this would lose the drill when trying to drill water
+        --       but I guess it would not work well with try which
+        --       might eventually be provided by some time-machine
```

Comment:
Agreed.  It would not break anything, but it does seem a bit weird to have an error be partly recoverable but also have irrecoverable consequences.